velox/exec/TableScan.cpp

```diff
@@ -138,6 +139,11 @@ bool TableScan::isFinished() {
 }
 
 void TableScan::setBatchSize() {
+  if (preferredBatchSize_ != 1024) {
```

Comment:
Do you have any suggestion to handle the relationship between the calculated batch size and the set one?